api.go

```diff
@@ -175,6 +184,71 @@ func EquipmentsHandler(manager *DataManager) gin.HandlerFunc {
 	}
 }
 
+func updateParameterTypes(param * FreeFloatingRequestParameter, types []string) {
+	for _, value := range types {
+		enumType := ParseFreeFloatingTypeFromParam(value)
+		if enumType != UnknownType {
+			param.types = append(param.types, enumType)
+		}
+	}
+}
+
+func initParameters(c *gin.Context) (param *FreeFloatingRequestParameter, err error) {
+	var longitude, latitude float64
+	var e error
+	p := FreeFloatingRequestParameter{}
+	countStr := c.DefaultQuery("count", "10")
+	p.count = stringToInt(countStr, 10)
+	distanceStr := c.DefaultQuery("distance", "500")
+	p.distance = stringToInt(distanceStr, 500)
+
+	types, _ := c.Request.URL.Query()["type[]"]
+	updateParameterTypes(&p, types)
+
+	coordStr := c.Query("coord")
+	if len(coordStr) == 0 {
+		err = fmt.Errorf("Bad request: coord is mandatory")
+		return nil, err
```

Comment:
You could just return the error, (and remove the error as a parameter) : 
```suggestion
		return nil, fmt.Errorf("Bad request: coord is mandatory")
```
Same apply below.